src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -169,14 +218,157 @@ export const selectRewardsByAccountSpecifier = createDeepEqualOutputSelector(
+    }
+
+    if (stakingData.rewards) {
+      initialValidatorsAddresses.push(...stakingData.rewards.map(x => x.validator.address))
+    }
+
+    const uniqueValidatorAddresses = [...new Set(initialValidatorsAddresses)]
+    return uniqueValidatorAddresses.filter(x => allValidators[x] === undefined)
+  }
+)
+
+export const selectUndelegationsAmountByValidatorAddress = memoize(
+  (
+    allUndelegationsEntries: Record<string, chainAdapters.cosmos.UndelegationEntry[]>,
+    validatorAddress: string
+  ) => {
+    return get(
+      allUndelegationsEntries,
+      validatorAddress,
+      [] as chainAdapters.cosmos.UndelegationEntry[]
+    )
+      .reduce((acc: BigNumber, undelegationEntry: chainAdapters.cosmos.UndelegationEntry) => {
+        acc = acc.plus(bnOrZero(undelegationEntry.amount))
+        return acc
+      }, bnOrZero(0))
+      .toString()
+  },
+  (
+    allUndelegationsEntries: Record<string, chainAdapters.cosmos.UndelegationEntry[]>,
+    validatorAddress: string
+  ) =>
+    get(allUndelegationsEntries, validatorAddress, [] as chainAdapters.cosmos.UndelegationEntry[])
+)
+
+export const selectRewardsAmountByValidatorAddress = memoize(
+  (allRewards: Record<string, chainAdapters.cosmos.Reward[]>, validatorAddress: string) => {
+    return get(allRewards, validatorAddress, [] as chainAdapters.cosmos.Reward[])
+      .reduce((acc: BigNumber, rewardEntry: chainAdapters.cosmos.Reward) => {
+        acc = acc.plus(bnOrZero(rewardEntry.amount))
+        return acc
+      }, bnOrZero(0))
+      .toString()
+  },
+  (allRewards: Record<string, chainAdapters.cosmos.Reward[]>, validatorAddress: string) =>
+    get(allRewards, validatorAddress, [] as chainAdapters.cosmos.Reward[])
+)
+
+export const selectTotalCryptoAmount = memoize(
+  (delegationsAmount: string, undelegationsAmount: string) => {
+    return bnOrZero(delegationsAmount).plus(bnOrZero(undelegationsAmount)).toString()
+  }
+)
+
+export const selectActiveStakingOpportunityDataByDenom = createDeepEqualOutputSelector(
+  selectAllDelegationsCryptoAmountByDenom,
+  selectAllUnbondingsEntriesByDenom,
+  selectAllRewardsByDenom,
+  selectAllValidators,
+  (
+    allDelegationsAmount,
+    allUndelegationsEntries,
+    allRewards,
+    allValidators
+  ): ActiveStakingOpportunity[] => {
+    return Object.entries(allValidators).reduce(
+      (acc: ActiveStakingOpportunity[], [validatorAddress, { apr, moniker }]) => {
+        const delegationsAmount = allDelegationsAmount[validatorAddress] ?? '0'
+
+        const undelegationsAmount = selectUndelegationsAmountByValidatorAddress(
+          allUndelegationsEntries,
+          validatorAddress
+        )
+
+        const rewards = selectRewardsAmountByValidatorAddress(allRewards, validatorAddress)
+
+        const cryptoAmount = selectTotalCryptoAmount(delegationsAmount, undelegationsAmount)
+
+        if (cryptoAmount !== '0' || rewards !== '0') {
```

Comment:
We shouldn't make check on numbers as strings and should BN them before. 
[`isZero`](https://mikemcl.github.io/bignumber.js/#isZ) can be used as a similar check, but from looking at this logic, we really want to check for numbers greater than 0, not negative values, which calls from [`gte`](https://mikemcl.github.io/bignumber.js/#gte).

Reply:
Fixed with `gt` to avoid having validators with 0 cryptoamount / rewards